src/main/java/com/faforever/client/chat/UserFilterController.java

```diff
@@ -124,23 +136,55 @@ boolean isGameStatusMatch(ChatUserItemController chatUserItemController) {
   }
 
   public void onGameStatusPlaying(ActionEvent actionEvent) {
-    playerStatusFilter = PLAYING;
-    gameStatusMenu.setText(i18n.get("game.gameStatus.playing"));
+    if (playerStatusFilter == PLAYING) {
+      updateGameStatusMenuText(null);
+    } else {
+      updateGameStatusMenuText(PLAYING);
+    }
     filterUsers();
   }
 
   public void onGameStatusLobby(ActionEvent actionEvent) {
-    playerStatusFilter = LOBBYING;
-    gameStatusMenu.setText(i18n.get("game.gameStatus.lobby"));
+    if (playerStatusFilter == LOBBYING) {
+      updateGameStatusMenuText(null);
+    } else {
+      updateGameStatusMenuText(LOBBYING);
+    }
     filterUsers();
   }
 
   public void onGameStatusNone(ActionEvent actionEvent) {
-    playerStatusFilter = IDLE;
-    gameStatusMenu.setText(i18n.get("game.gameStatus.none"));
+    if (playerStatusFilter == IDLE) {
+      updateGameStatusMenuText(null);
+    } else {
+      updateGameStatusMenuText(IDLE);
+    }
```

Comment:
You are doing this 3 times in a row, pls extract the if else logic into a method (e.g. updateGameStatusMenuText).

Reply:
I did, but then @micheljung requested this:

>I'd rather like to see the call like this:
>
>if(((Toggle) actionEvent.getTarget()).isSelected()) {
>  setGameStatusFilter(IDLE);
>} else {
>  setGameStatusFilter(null);
>}
>
>Yes, it's more code and less fancy, but very explicit and thus easy to understand, no magic >involved.

Reply:
I see now. SetOrUnset... is a very confusing method name, I agree, rather choose sth like update or so. But if @micheljung said so then it should be fine, is a bit overkill but makes it more readable.

Reply:
I agree, having the if-else 3 times is ugly. How about this?

```
updateGameStatusMenuText(playerStatusFilter == IDLE ? null : IDLE);
```